active/0002-centalized-mgmt-resource-types-provider-versions.md

```diff
@@ -0,0 +1,259 @@
+
+1. The new syntax is quite verbose, and cannot be controlled centrally. For users with a large set of files, picking up the latest version of `az` will require modifying every single file.
+1. The aliasing (assuming that the last segment of the registry namespace == the provider name) & versioning (having an `@` character instead of a `:` which is familiar to module users) behavior of the new syntax can be confusing.
+1. Its deceiving and confusing to allow `sys` to have a pinned versionis fixed version `1.0.0` - this **should** imply that there are no changes between Bicep versions, which isn't realistic.
+
+- It's an "either/or" - either users follow the current behavior (`az` imported by default) **or** they're using the new syntax. We don't have a plan for how to combine the two.
+- With [Radius](https://github.com/radius-project/radius), it's conceivable that we may want users to **not** have `az` imported by default.
+
+### Proposed Changes
+
+- [Title - Centralized management of resource type provider versions using package.json](#title---centralized-management-of-resource-type-provider-versions-using-packagejson)
+  - [Summary](#summary)
+  - [Terms and definitions](#terms-and-definitions)
+  - [Motivation](#motivation)
+    - [Proposed Changes](#proposed-changes)
+  - [Detailed design](#detailed-design)
+    - [Client side changes](#client-side-changes)
+      - [Add a new syntax for the provider import declaration statements: `import {providerName} [as {optionalAlias}]`](#add-a-new-syntax-for-the-provider-import-declaration-statements-import-providername-as-optionalalias)
+      - [Remove support for syntax `import '{providerName}@{providerVersion}' [as {optionalAlias}]`](#remove-support-for-syntax-import-providernameproviderversion-as-optionalalias)
+      - [Change the delimiter '@' used by the supported provider declaration syntax to ':' to align with module reference syntax and increase consistency](#change-the-delimiter--used-by-the-supported-provider-declaration-syntax-to--to-align-with-module-reference-syntax-and-increase-consistency)
+  - [Drawbacks](#drawbacks)
+  - [Alternatives](#alternatives)
+  - [Rollout plan](#rollout-plan)
+  - [Unresolved questions](#unresolved-questions)
+  - [Out of scope](#out-of-scope)
+
+## Detailed design
+
+### Client side changes
+
+#### Add a new syntax for the provider import declaration statements: `import {providerName} [as {optionalAlias}]`
+
+The example below shows a usage of the proposed syntax
+
+```bicep
+provider az
+provider kubernetes as k8s // uses optional aliasing of the
+```
+
+The resolution of the symbols will be resolved from the builtin `bicepconfig.json` under a new section:
+
+```json
+{
+  // prior bicepconfig.json sections
+  // ...
+
+  "providers": {
+    "az": {
+      // A package reference string (below) gestures Bicep to source the types from the registry address.
+      "source": "mcr.microsoft.com/bicep/providers/az:0.2.3", 
```

Comment:
In the 99% case, types are going to be pulled from the public registry. Can we provide an option for only adjusting the version of az instead of the full or aliased registry path?

Reply:
Let's look out for costumers' feedback on the first iteration and discuss this as part of GA release

Reply:
I decided to reduce duplicate names and simplify this to the following:
```json
{
    "providers":{
        "az":{
            "source":  "mcr.microsoft.com/bicep/providers",
            "version: "1.2.3",
            "isDefaultImport": true
        }
    }
}
```